torch/csrc/autograd/saved_variable.cpp

```diff
@@ -214,8 +210,8 @@ Variable SavedVariable::unpack(std::shared_ptr<Node> saved_for) const {
     var = make_variable(data, requires_grad_);
   }
 
-  impl::set_version_counter(var, version_counter_);
   impl::set_grad_accumulator(var, grad_accumulator_);
+  impl::set_version_counter(var, impl::version_counter(data));
```

Comment:
The reason we DO still need to do some save/restore here is because make_variable resets the version counter to 0.